classes/sota_raspberrypi.bbclass

```diff
@@ -1,11 +1,20 @@
 RPI_USE_U_BOOT_sota = "1"
-KERNEL_IMAGETYPE_sota = "uImage"
+
+KERNEL_CLASSES_append_sota = " kernel-fitimage"
+KERNEL_IMAGETYPE_sota = "fitImage"
```

Comment:
This will force fitImage for all rpi targets, and it will also require the extra patch that is available at meta-updater-raspberrypi (which seems to be changing only rpi_3_32b_defconfig atm).

I'm curious if we should move most of this logic into that layer instead, as it is a bit weird to have some hw support here and the rest in another layer, and you can't really use one without the other.

Reply:
@rsalveti Historically meta-updater was developed in AGL codebase and only later we decided to create a stand-alone environment with updater-repo. When we did it it turned out that AGL's platform-specific layers contained some useful paches that we needed for standalone use case as well. But as meta-updater should still have been compatible with AGL, they couldn't have been added to meta-updater itself, hence meta-updater-<platform> layers.
And still we want to control most of platform-specific logic inside meta-updater itself, because every change to meta-updater-<platform> has to be mirrored to meta-agl-bsp/<platform> and I'd like to minimise this effort.
Looks ugly, I agree, but I couldn't come up with a better solution yet.

Reply:
@rsalveti Your note about support for raspberrypi 2 is totally legit, I think we should either support it properly or explicitly only support RPi3

Reply:
> Looks ugly, I agree, but I couldn't come up with a better solution yet.

Got it, thanks for explaining the background.

> I think we should either support it properly or explicitly only support RPi3

I would be fine supporting just rpi3, as it is now the default rpi target anyway.